mcs/class/Mono.Security/Mono.Security/StrongName.cs

```diff
@@ -422,7 +535,6 @@ public bool Sign (string fileName)
 			using (FileStream fs = File.OpenWrite (fileName)) {
 				fs.Position = sn.SignaturePosition;
 				fs.Write (signature, 0, signature.Length);
-				fs.Close ();
 				result = true;
```

Comment:
Just `return true` and get rid of the `result` variable?

Reply:
I was wondering about that. In msvc C at least, you don't want to return from a __try, better to fall out. Return within __try runs exception handling code. I wasn't sure if that thinking was being applied here.

Reply:
Just for the record, the compiler moves the return out. Consider the following C# code:

```csharp
int Test ()
{
	using (System.IO.File.OpenRead("aaa.text"))
		return 0;
}
```

translates into the following IL:

```
.method private hidebysig 
	instance int32 Test () cil managed 
{
	// Method begins at RVA 0x2050
	// Code size 29 (0x1d)
	.maxstack 1
	.locals init (
		[0] class [mscorlib]System.IO.FileStream,
		[1] int32
	)

	IL_0000: nop
	IL_0001: ldstr "aaa.text"
	IL_0006: call class [mscorlib]System.IO.FileStream [mscorlib]System.IO.File::OpenRead(string)
	IL_000b: stloc.0
	.try
	{
		IL_000c: ldc.i4.0
		IL_000d: stloc.1
		IL_000e: leave.s IL_001b
	} // end .try
	finally
	{
		IL_0010: ldloc.0
		IL_0011: brfalse.s IL_001a

		IL_0013: ldloc.0
		IL_0014: callvirt instance void [mscorlib]System.IDisposable::Dispose()
		IL_0019: nop

		IL_001a: endfinally
	} // end handler

	IL_001b: ldloc.1
	IL_001c: ret
} // end of method Program::Test
```

I'm not Jon Skeet to link to all the technical details in the specification, but I am sure that if the compiler considers it the same then it should be semantically equivalent. :-)